pre_commit/languages/docker.py

```diff
@@ -10,6 +10,7 @@
 from pre_commit.util import CalledProcessError
 from pre_commit.util import cmd_output_b
 
+ENV_VAR_OVERRIDE_HOST_PATH = 'PRE_COMMIT_OVERRIDE_DOCKER_HOST_PATH'
```

Comment:
this doesn't actually implement what the title of this PR says -- I don't really like sprinkling environment variable hacks so I can't accept this as it is.  I would accept a PR which improves the detection -- but as it is right now this is just a workaround that I'd have to document and maintain forever rather than an actual solution

Reply:
That was the intent of my original PR yes. 

Unfortunately, after reading and testing I realized there isn't a consistent way of obtaining the container ID that works on all platforms for version 2 of cgroups...  I opted for the simpler solutions of an env var as that has the benefit on consistent functionality.